### ISSUE TITLE
Random text input from mouse actions

### ISSUE THREAD

**Issue by zareami10** [zellij-org/zellij#932]
**Basic information**

`zellij --version`: 0.21.0
`OS:` macOs Monterey

<img width="1658" alt="Screen Shot 2021-12-09 at 21 44 46" src="https://user-images.githubusercontent.com/59122438/145400222-3193d7ef-592e-427f-a5c3-3b4aca58bf5a.png">

This issue starts (apparently) randomly, but once it starts I can not actually stop it , and scrolling/entering commands becomes impossible as more input gets written and terminal scrolls randomly. The issue happens with multiple terminal emulators including WezTerm and default mac terminal.

Unfortunately I can not yet exactly reproduce the issue (which might have something to do with mouse reporting getting enabled?), but please let me know if there is any further information I can provide.

**Comment by imsnif** [comment]
@tlinford - do these look like scroll instructions to you?

**Comment by tlinford** [comment]
@imsnif yep it sure looks like it -  seems to be either mode 1006 or 1015 by looking at https://invisible-island.net/xterm/ctlseqs/ctlseqs.html#h3-Extended-coordinates

**Comment by imsnif** [comment]
Thanks @tlinford !

@zareami10 - are you running with `--disable-mouse-mode` by any chance? (this is still a bug in any case, just trying to pin-point why this is happening)

**Comment by zareami10** [comment]
> @zareami10 - are you running with `--disable-mouse-mode` by any chance? (this is still a bug in any case, just trying to pin-point why this is happening)

No, I have not changed any settings. And I think [the Formula](https://github.com/Homebrew/homebrew-core/blob/HEAD/Formula/zellij.rb) which I have used for installation doesn't have any customization either.

Maybe I should mention that the issue also happens when I run Zellij on a Linux remote machine (with the host still being macOS).

**Comment by imsnif** [comment]
@tlinford - any ideas? (if you have time to take a look)

**Comment by tlinford** [comment]
@imsnif for now I tried reproducing on wezterm with a touchpad (on linux), but no luck.
@zareami10 just to be sure - could you try running `zellij options --disable-mouse-mode` and see if you still get the problem?

**Comment by zareami10** [comment]
> @imsnif for now I tried reproducing on wezterm with a touchpad (on linux), but no luck. @zareami10 just to be sure - could you try running `zellij options --disable-mouse-mode` and see if you still get the problem?

I've also so far been unable to reproduce the issue on Linux.
When I use the `--disable-mouse-mode` option, when the bug is triggered there are no random inputs as before, but Zellij keeps scrolling through ZSH history instead on scroll events.

**Comment by imsnif** [comment]
Does the ZSH history scrolling happen outside of Zellij too?

**Comment by tlinford** [comment]
@imsnif I'm a bit stumped to be honest - regarding the problem observed with mouse mode on, as far as I can tell those all look like mouse release events, except they are bypassing termion for some reason?

**Comment by imsnif** [comment]
@tlinford - could it be that we're somehow doing the wrong thing here before sending stuff to termion (here we essentially deal with bracketed paste and home/end keys in terminfo): https://github.com/zellij-org/zellij/blob/main/zellij-client/src/stdin_handler.rs#L63-L119 ?

@zareami10 - if this doesn't happen in ZSH outside of Zellij, any chance you could check in version `0.19.0` and see if it still happens there?

**Comment by zareami10** [comment]
> Does the ZSH history scrolling happen outside of Zellij too?

No, it seems to be limited to Zellij.

> @zareami10 - if this doesn't happen in ZSH outside of Zellij, any chance you could check in version `0.19.0` and see if it still happens there?

Unfortunately the issue also happens on 0.19.0.

**Comment by tlinford** [comment]
@zareami10 would you be comfortable building and running from a debug branch and provide us some logs to try and figure out what is going wrong?
If so, you can check out and build branch https://github.com/zellij-org/zellij/tree/debug/%23932-mouse-events-stdin, and share the log file after you hit the issue.
Please not that this will be logging all key presses you enter so I recommend not writing anything sensitive :)

**Comment by raphCode** [comment]
Holding Ctrl or Shift while using the mouse also leads to mouse escape sequences passed plaintext through to the program running in zellij, maybe this is related?

`--disable-mouse-mode` also stops this behavior

**Comment by zareami10** [comment]
> @zareami10 would you be comfortable building and running from a debug branch and provide us some logs to try and figure out what is going wrong? If so, you can check out and build branch https://github.com/zellij-org/zellij/tree/debug/%23932-mouse-events-stdin, and share the log file after you hit the issue. Please not that this will be logging all key presses you enter so I recommend not writing anything sensitive :)

I have attached the [zellij-log](https://github.com/zellij-org/zellij/files/7717250/zellij-log.zip) folder.

**Comment by zareami10** [comment]
> @raphCode are you also seeing the problem on macOS?

I can reproduce a similar behavior, but I thought that one was intentional, as it happens regardless of the OS and terminal, unlike this bug.
Basically holding ctrl, alt or shift and scrolling would type out the scroll events in Zellij. Somewhat similar to this bug, but it wouldn't get out of control unlike this one.

**Comment by raphCode** [comment]
> @raphCode are you also seeing the problem on macOS?

I have only tested on linux.

> I can reproduce a similar behavior, but I thought that one was intentional

I also thought that, but since that behavior disappears with `--disable-mouse-mode` I am not so sure anymore...

**Comment by tlinford** [comment]
Thanks to both of you! I'm now able at least to reproduce the text input appearing - the characters were not showing up with fish, but trying it out in bash they are printed out.

**Comment by tlinford** [comment]
Hi @zareami10, we would really like to get to the bottom of this problem, would you be patient and give us some more details? 🙂

With mouse mode enabled, when you say that the terminal scrolls randomly, what do you mean exactly?
Is it scrolling down because new input appears continuously, or moving up and down randomly?

In the other situation, with mouse-mode turned off, is the scrolling through zsh history random, or does it simply do it on scroll up/down?

**Comment by zareami10** [comment]
> Hi @zareami10, we would really like to get to the bottom of this problem, would you be patient and give us some more details? 🙂

Sure.

> With mouse mode enabled, when you say that the terminal scrolls randomly, what do you mean exactly? Is it scrolling down because new input appears continuously, or moving up and down randomly?

Once the bug is triggered, you actually can scroll if you are careful not to move the mouse pointer. But if the pointer moves slightly new inputs appear and you're returned to the bottom.

> In the other situation, with mouse-mode turned off, is the scrolling through zsh history random, or does it simply do it on scroll up/down?

It is a normal scrolling behaviour going up/down depending on the scroll direction. It also has the "kinetic" effect enabled, if it makes any sense in such a situation.

If it would help I could provide a recording too, so just let me know.

**Comment by tlinford** [comment]
thanks, that's very helpful - i think what you are seeing with mouse mode off is faux scrolling, I found out about it here: https://github.com/alacritty/alacritty/issues/1194

**Comment by tlinford** [comment]
@zareami10 i pushed a new commit on the branch you tested on, to workaround this problem and avoid spamming all the mouse events to the terminal. Could you test it out and let me know?

**Comment by zareami10** [comment]
> @zareami10 i pushed a new commit on the branch you tested on, to workaround this problem and avoid spamming all the mouse events to the terminal. Could you test it out and let me know?

With the new commit I am no longer able to reproduce the mouse event inputs. But one issue still remains; using swipe left/right (even slightly) with the touch-pad keeps copying the text to clipboard (which is reflected on the status bar). One detail I have noticed is that text keeps getting copied over and over even in a single swipe until you stop the "gesture".

**Comment by tlinford** [comment]
Got it, my guess is that happens because during the gesture some mouse release events are being sent, and we currently don't track what button was pressed.

**Comment by tlinford** [comment]
@zareami10  - if you would be so patient, when you get a chance there's another commit on the branch to try that should avoid the copying while swiping.

**Comment by zareami10** [comment]
> @zareami10 - if you would be so patient, when you get a chance there's another commit on the branch to try that should avoid the copying while swiping.

Regarding the mouse issues, things are looking good so far!
I'm not sure if this is related, but I noticed that extra line when opening Zellij from your branch, which doesn't happen in the released version:

<img width="686" alt="Screen Shot 2021-12-18 at 12 14 10" src="https://user-images.githubusercontent.com/59122438/146627202-72674255-0f3f-4e7f-a89f-59b568a9c8a8.png">

**Comment by tlinford** [comment]
Do you get the extra line if you try running the latest commit in main?

**Comment by zareami10** [comment]
> Do you get the extra line if you try running the latest commit in main?

I was able to reproduce it once on main, so I guess it's not related to your branch. But it's quite hard to reproduce sometimes so I can't say if the issue is new or not...

**Comment by tlinford** [comment]
Alright, good to know - if you want feel free to open up a separate issue for that, and thanks for helping to solve this problem.

**Comment by smileart** [comment]
I know this one was closed, but I still have this issue, so I wonder if I've misconfigured something or is it reproducible with my combination of tools/environment. Using Kitty 0.26.5 with Zellij 0.34.4 and zsh 5.8.1 under macOS 13.0.1 (M1).

In pure ZSH terminal without a Zellij session there are no issues whatsoever, but when I scroll (using my laptop's trackpad) in Zellij's pane I always get these...

<img width="879" alt="image" src="https://user-images.githubusercontent.com/455338/209351955-bfcad1b7-02a8-4327-bf66-6dd9ded17360.png">

**Comment by tlinford** [comment]
Hey @smileart, you are most likely hitting #1716 - I hope to fix it sometime soon. In the meanwhile you could get around it by using a different terminal emulator that doesn't send horizontal scroll events.

**Comment by smileart** [comment]
@tlinford thanks for the prompt response! Yes, it seems to be exactly the case. Somehow I didn't find that one. Gonna track the progress there. Cheers! 🤝

**Comment by jfboisvieux** [comment]
I do have the same problem when I scroll using Kitty and Zellij, 
this does not happen using iterm2 on my mac, but I need Kitty since I'm editing with Hellix (hx) and Kitty allows me to get a proper Alt 

is there any workaround ?

**Comment by smileart** [comment]
@jfboisvieux I ended up scrolling only vertically (with a mouse, with Homerow.app and with Karabiner shortcuts to control the mouse pointer, while still using Kitty. So far, so good. While the issue gets fixed.

**Comment by jfboisvieux** [comment]
Good to know tks !___##   Ipad emailJean-Francois Boisvieuxmobile : (33) 6 32 49 26 98email : ***@***.*** 28 Jan 2023, at 12:18, Serge Bedzhyk ***@***.***> wrote:﻿
@jfboisvieux I ended up scrolling only vertically (with a mouse, with Homerow.app and with Karabiner shortcuts to control the mouse pointer, while still using Kitty. So far, so good. While the issue gets fixed.

**Comment by thisismygitrepo** [comment]
Can someone please reopen this? This still happens sometimes on Feb 2024 version. It is break my SSH sessions as the server can't handle this influx of input.

**Comment by mariusrueve** [comment]
Still happens to me on Ubuntu 24.04 using both Gnome Terminal and Ghostty. Happens when I Tab back to the terminal window.

**Comment by jhedlund** [comment]
Having this issue on Windows Terminal, only on a Surface Pro trackpad, and it seems to be only when I move the cursor between panes - the panes start to go to different places.

**Comment by jhedlund** [comment]
Further investigation, it's not just between panes, I can just move the pointer within one pain in random directions and it's basically calling the Alt-[ ] and changing the orientation of the panes.   I also get text like this in a pane:  35;107;13m35;106;13m35;105;13m35;104;13m35;114;11m35;112;11m35;110;11m35;109;11m5;73;11m35;71;11m35;70;11m5;87;13m35;89;13m35;90;13m35;92;14m;90;6m35;89;7m35;89;8m35;88;9m5;88;25m35;87;26m35;86;27m35;85;28m

Let me know if I should open a new issue, but seems related to this issue.

**Comment by celynw** [comment]
I have this problem and it's reproducible:

1. Outside of `zellij`, SSH to another device
2. Attach to `zellij` session on remote device
3. Interrupt the SSH session. Most commonly by closing my laptop lid, which closes my VPN session
4. The terminal window now shows this behaviour. I get them from moving or scrolling on the mouse and trackpad.

This happens in `ghostty` and `gnome-terminal`.

* `zellij 0.42.2`
* `uname -av`: `Linux hostname 6.8.0-59-generic #61~22.04.1-Ubuntu SMP PREEMPT_DYNAMIC Tue Apr 15 17:03:15 UTC 2 x86_64 x86_64 x86_64 GNU/Linux`

I considered to open a new issue but it seemed too close to this one.

**Comment by fizzxed** [comment]
> this one is unrelated. it happens because zellij enables mouse movement handling and doesnt disable it when you close the connection. i am not sure if this is even solvable in zellij. i will probably investigate it on saturday

Can emulate this issue by sending an ssh escape sequence`~.` to close the session when zellij is running. Were you able to find out if this is at all solvable on the zellij side? Happens quite often if one is on the move and doesn't detach first, or if the connection is spotty.

**Comment by nowave7** [comment]
I'm also experiencing this issue. My setup is as follows:
- WindowsTerminal app on Windows 11 laptop
  (but it also happens with wezterm or putty)
- ssh into a remote Ubuntu Linux machine using zsh
- run zellij attach session-number
- just move the mouse around (I don't use trackpad):
- sometimes it happens only after ~15 seconds of moving the mouse
- sometimes it is on every mouse action

zellij version:
zellij 0.43.1
zsh version:
zsh 5.9 (x86_64-ubuntu-linux-gnu)
Let me know if more info is needed.

<img width="707" height="120" alt="Image" src="https://github.com/user-attachments/assets/b913d692-5d65-4060-8b10-352450a5b075" />

**Comment by XC3N** [comment]
Hello, same thing for me on windows 11 using  Alacritty...

Triggered by both the horizontal mouse scroll on my logitech MX master 3s, and sporadically when the mouse enters or quits the terminal window

**Comment by divens** [comment]
> Hello, same thing for me on windows 11 using Alacritty...
> 
> Triggered by both the horizontal mouse scroll on my logitech MX master 3s, and sporadically when the mouse enters or quits the terminal window

Hello @XC3N , I assume you built from main? Can you tell me exactly which commit? Thanks!

Maybe I misread. Is it happening exactly as with previous message, i.e. when ssh-ing from zsh on Windows (I assume WSL/Ubuntu)? Or is it related to a native Windows build?

**Comment by XC3N** [comment]
> > Hello, same thing for me on windows 11 using Alacritty...
> > Triggered by both the horizontal mouse scroll on my logitech MX master 3s, and sporadically when the mouse enters or quits the terminal window
> 
> Hello [@XC3N](https://github.com/XC3N) , I assume you built from main? Can you tell me exactly which commit? Thanks!
> 
> Maybe I misread. Is it happening exactly as with previous message, i.e. when ssh-ing from zsh on Windows (I assume WSL/Ubuntu)? Or is it related to a native Windows build?

Thanks for getting back to me!

Sorry for the lack of details. I have this issue on two different servers running nixos, both on 0.43.1 from nixpkgs
The issue is reproduced from two different win11 clients, using alacritty to SSH into either server
Symptoms seem worse with my logitech mx master 3s, because it has a thumbwheel for sidescrolling, but happens with a regular three buttons mouse (sporadically while moving the mouse cursor in and out of frame of alacritty, I get stuff like 5;19;33M5;51;35M5;20;37M5;34;43M)

Let me know if I can clarify anything or perform tests to thelp

**Comment by divens** [comment]
> Thanks for getting back to me!
> 
> Sorry for the lack of details. I have this issue on two different servers running nixos, both on 0.43.1 from nixpkgs The issue is reproduced from two different win11 clients, using alacritty to SSH into either server Symptoms seem worse with my logitech mx master 3s, because it has a thumbwheel for sidescrolling, but happens with a regular three buttons mouse (sporadically while moving the mouse cursor in and out of frame of alacritty, I get stuff like 5;19;33M5;51;35M5;20;37M5;34;43M)
> 
> Let me know if I can clarify anything or perform tests to thelp

I can reproduce this easily with zellij v0.43.1

Good news is that upcoming v0.44.0 (current main) is much, much less prone to this problem. Bad news is that it does still happen from time to time. I have not come up yet with an easy way to reproduce. I'll keep digging.
